deps/rabbit/test/amqp_client_SUITE.erl

```diff
@@ -6596,8 +6597,55 @@ bad_x_cc_annotation_exchange(Config) ->
     ok = end_session_sync(Session),
     ok = close_connection_sync(Connection).
 
+%% Attach a receiver to an unavailable quorum queue.
+attach_to_down_quorum_queue(Config) ->
+    QName = <<"q-down">>,
+    Address = rabbitmq_amqp_address:queue(QName),
+
+    %% Create quorum queue with single replica on node 2.
+    {_, _, LinkPair2} = Init2 = init(2, Config),
+    {ok, _} = rabbitmq_amqp_client:declare_queue(
+                LinkPair2,
+                QName,
+                #{arguments => #{<<"x-queue-type">> => {utf8, <<"quorum">>},
+                                 <<"x-quorum-initial-group-size">> => {ulong, 1}
+                                }}),
+    ok = close(Init2),
+
+    %% Make quorum queue unavailable.
+    ok = rabbit_ct_broker_helpers:stop_broker(Config, 2),
+
+    OpnConf = connection_config(0, Config),
+    {ok, Connection} = amqp10_client:open_connection(OpnConf),
+    {ok, Session0} = amqp10_client:begin_session_sync(Connection),
+    flush(attaching_receiver),
+    {ok, _Receiver} = amqp10_client:attach_receiver_link(
+                        Session0, <<"receiver">>, Address),
+    receive
+        {amqp10_event,
+         {session, Session0,
+          {ended,
+           #'v1_0.error'{
+              condition = ?V_1_0_AMQP_ERROR_INTERNAL_ERROR,
+              description = {utf8, Desc}}}}} ->
+            ?assertEqual(
+               <<"failed consuming from quorum queue 'q-down' in vhost '/': noproc">>,
```

Comment:
Should we replace the "noproc" with something like "the leader replica was unavailable or not elected"?

Reply:
Ideally yes. However, `noproc` is an exit reason returned from https://www.erlang.org/doc/apps/stdlib/gen_statem.html#call/3. That function can return other exit reasons too. I'd like to avoid parsing and translating any possible exit reason from `gen_statem:call/3`. The alternative would be to log this Erlang term only in the server and not pass it down to the client app.

FWIW, I removed the `noproc` from the test expectation.